x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts

```diff
@@ -476,6 +477,17 @@ export class SecureSavedObjectsClientWrapper implements SavedObjectsClientContra
     return await this.redactSavedObjectsNamespaces(response);
   }
 
+  public async removeReferencesTo(
+    type: string,
+    id: string,
+    options: SavedObjectsRemoveReferencesToOptions = {}
+  ) {
+    const args = { type, id, options };
+    await this.ensureAuthorized(type, 'delete', options.namespace, { args });
```

Comment:
We'll want to include audit logging for this operation. Here's a snippet to get started, but it'll need the requisite changes inside `audit_events`. The changes within `audit_events` should be mostly straightforward, but let me know if you need any help with it!

```suggestion
    try {
      await this.ensureAuthorized(type, 'delete', options.namespace, {
        args,
        auditAction: 'removeReferences',
      });
    } catch (error) {
      this.auditLogger.log(
        savedObjectEvent({
          action: SavedObjectAction.REMOVE_REFERENCES,
          savedObject: { type, id },
          error,
        })
      );
    }

    this.auditLogger.log(
      savedObjectEvent({
        action: SavedObjectAction.REMOVE_REFERENCES,
        savedObject: { type, id },
        outcome: EventOutcome.UNKNOWN,
      })
    );
```

Reply:
Done in https://github.com/elastic/kibana/pull/79096/commits/8a24f1f2c4c4007524432f8c7429decaf5d18419, please tell me what you think.

- Setting the `eventVerb` was a little awkward, as I needed more than a verb to be explicit, but that logs nicely. Tell me if you think I should use something else

- Not sure of the value for `eventTypes` either, as it's kinda of a special case (we are not updating the actual object the operation is logged for). I used `DELETION`, but maybe you may want `CHANGE` instead?

Reply:
> Setting the eventVerb was a little awkward, as I needed more than a verb to be explicit, but that logs nicely. Tell me if you think I should use something else

I like what you've done here.

> Not sure of the value for eventTypes either, as it's kinda of a special case (we are not updating the actual object the operation is logged for). I used DELETION, but maybe you may want CHANGE instead?

Yeah, I think `CHANGE` makes more sense than `DELETION` for this particular operation. We will already log a `DELETION` event when the tag object itself is removed.

Reply:
cc @thomheymann for visibility

Reply:
Used `CHANGE` instead (https://github.com/elastic/kibana/pull/79096/commits/5f66b49f5d2ddc0c40723b143a0faf085d450f56)